### PR TITLE
Linguistic Probing Tasks

### PR DESCRIPTION
**Pull request by asparius**
## Checklist for adding MMTEB dataset

Reason for dataset addition:

It is a linguistic probing tasks proposed in this [paper](https://arxiv.org/abs/1805.01070). We could have them as a new Task however they are classification tasks at the end.

- [x] I have tested that the dataset runs with the `mteb` package.
- [x] I have run the following models on the task (adding the results to the pr). These can be run using the `mteb run -m {model_name} -t {task_name}` command.
  - [x] `sentence-transformers/paraphrase-multilingual-MiniLM-L12-v2`
  - [x] `intfloat/multilingual-e5-small`
- [x] I have checked that the performance is neither trivial (both models gain close to perfect scores) nor random (both models gain close to random scores).
- [x] If the dataset is too big (e.g. >2048 examples), considering using `self.stratified_subsampling() under dataset_transform()`
- [x] I have filled out the metadata object in the dataset file (find documentation on it [here](https://github.com/embeddings-benchmark/mteb/blob/main/docs/adding_a_dataset.md#2-creating-the-metadata-object)).
- [x] Run tests locally to make sure nothing is broken using `make test`. 
- [x] Run the formatter to format the code using `make lint`. 
- [ ] I have added points for my submission to the [points folder](https://github.com/embeddings-benchmark/mteb/blob/main/docs/mmteb/points.md) using the PR number as the filename (e.g. `438.jsonl`).

### CODE REVIEW

File: mteb/tasks/Classification/eng/LinguisticProbing.py [hunk 1]
```diff
@@ -0,0 +1,454 @@
...
+    )
+
+    def dataset_transform(self):
+        self.dataset = self.dataset.rename_column("sentence", "text")
+        self.dataset = self.stratified_subsampling(
+            self.dataset, seed=self.seed, splits=["test"]
+        )
+
+
+class WCClassification(AbsTaskClassification):
```
Comment (x-tabdeveloping) [inline mteb/tasks/Classification/eng/LinguisticProbing.py:52]: Correct me if I'm wrong but this does 1000-way classification, and my guess would be that it would take a lot of time and examples to train a classification head. Perhaps a keyword retrieval task would do the same trick way faster.

**Review by x-tabdeveloping — COMMENTED** [review]
Looking good! We should probably do sth about "topic classification". I was thinking some of the tasks might be easier to formulate as other tasks.
How long did it take to run the tasks?

File: mteb/tasks/Classification/eng/LinguisticProbing.py [hunk 2]
```diff
@@ -0,0 +1,454 @@
...
+        },
+        type="Classification",
+        category="s2s",
+        eval_splits=["test"],
+        eval_langs=["eng-Latn"],
+        main_score="accuracy",
+        date=("2018-01-01", "2018-07-08"),
+        form=["written"],
+        domains=["Non-fiction", "Fiction"],
+        task_subtypes=["Topic classification"],
```
[the remark this answers is not in the document]
Comment (asparius) [reply mteb/tasks/Classification/eng/LinguisticProbing.py:25]: Agree, but this is the only subtask that casts classification into it since probing tasks are classification basically

File: mteb/tasks/Classification/eng/LinguisticProbing.py [hunk 3]
```diff
@@ -0,0 +1,454 @@
+from __future__ import annotations
+
+from mteb.abstasks import AbsTaskClassification
+from mteb.abstasks.TaskMetadata import TaskMetadata
+
+
+class SentLenClassification(AbsTaskClassification):
```
[the remark this answers is not in the document]
Comment (asparius) [reply mteb/tasks/Classification/eng/LinguisticProbing.py:7]: I think classification is fine because sentence lengths are eventually like class labels. Regression is more difficult than classification with deep neural networks.

File: mteb/tasks/Classification/eng/LinguisticProbing.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (x-tabdeveloping) [reply mteb/tasks/Classification/eng/LinguisticProbing.py:25]: Well we can make one more, like idk: "Syntactic classification" or something

File: mteb/tasks/Classification/eng/LinguisticProbing.py (hunk 1 above)
Comment (asparius) [reply mteb/tasks/Classification/eng/LinguisticProbing.py:52]: Probably, it would work better but I stick with the paper's choice. The better option is to treat the probing task as a new dataset and we can change whatever we want

File: mteb/tasks/Classification/eng/LinguisticProbing.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (x-tabdeveloping) [reply mteb/tasks/Classification/eng/LinguisticProbing.py:7]: How is it more difficult? You just take the embeddings and train a regression head, LR could be fine but if we feel spicy could do an `MLPRegressor` with one hidden layer. I'm also thinking it would make a bit more sense, hence it's good to be able to __derive__ linguistic properties from embeddings, but good embeddings should _mostly_ encode semantics.
I would expect a kNN classifier to do waaay better on semantics, and not so good on syntax even when the model encodes it well.

File: mteb/tasks/Classification/eng/LinguisticProbing.py (hunk 1 above)
Comment (x-tabdeveloping) [reply mteb/tasks/Classification/eng/LinguisticProbing.py:52]: Hmmm, I think from a validity standpoint of view it's fine to stick with the paper's choice.
I'm just a bit worried that it would take a very long time to run.

File: mteb/tasks/Classification/eng/LinguisticProbing.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (x-tabdeveloping) [reply mteb/tasks/Classification/eng/LinguisticProbing.py:7]: In any case, I would probably change this task to use logistic regression instead of kNN, that's probably the least painful option :)

[the remark this answers is not in the document]
Comment (asparius) [reply mteb/tasks/Classification/eng/LinguisticProbing.py:7]: They provide results for logistic regression in the paper as well.

**Comment by asparius** [comment]
I will soon create an issue regarding concerns over this PR.

File: mteb/tasks/Classification/eng/LinguisticProbing.py [hunk 4]
```diff
@@ -0,0 +1,454 @@
...
+        },
+        type="Classification",
+        category="s2s",
+        eval_splits=["test"],
+        eval_langs=["eng-Latn"],
+        main_score="accuracy",
+        date=("2018-01-01", "2018-07-08"),
+        form=["written"],
+        domains=["Non-fiction", "Fiction"],
+        task_subtypes=["Topic classification"],
```
[the remark this answers is not in the document]
Comment (asparius) [reply mteb/tasks/Classification/eng/LinguisticProbing.py:115]: They have subtypes of the tasks in the paper, I will update metadatas accordingly.

File: mteb/tasks/Classification/eng/LinguisticProbing.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (x-tabdeveloping) [reply mteb/tasks/Classification/eng/LinguisticProbing.py:7]: Oh sorry I thought kNN was the default classification method on the AbsTask, don't worry about it then.

File: mteb/tasks/Classification/eng/LinguisticProbing.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (x-tabdeveloping) [reply mteb/tasks/Classification/eng/LinguisticProbing.py:115]: Jolly!

**Comment by KennethEnevoldsen** [comment]
Seems like this was moved to a discussion that was never finished. Will close it for now, but feel free to re-open it.
